File: frontend/src/api/menu.js
import request from '@/utils/request'

export const menu = {
  menuId: null,
  menuCode: null,
  menuName: null,
  comment: null,
  status: 1,
  hidden: false,
  fullScreen: false,
  level: null,
  sortNum: null,
  icon: null,
  parentId: null
}

export function findOneMenu(deptId) {
  return request({
    url: '/menu/' + deptId,
    method: 'get'
  })
}

export function findAllMenu(params) {
  return request({
    url: '/menu',
    method: 'get',
    params: params
  })
}

export function insertMenu(menu) {
  return request({
    url: '/menu',
    method: 'post',
    data: menu
  })
}

export function updateMenu(menu) {
  return request({
    url: '/menu',
    method: 'put',
    data: menu
  })
}

export function findMenuTree() {
  return request({
    url: '/menu/tree',
    method: 'get'
  })
}

export function getParentId(menuId) {
  return request({
    url: '/menu/getParentId',
    method: 'get',
    params: { menuId: menuId }
  })
}

export function deleteMenu(deptId) {
  return request({
    url: '/menu/' + deptId,
    method: 'delete'
  })
}

File: backend/core/src/main/java/com/zsf/core/entity/Menu.java
package com.zsf.core.entity;

import com.zsf.core.config.hibernate.SnowFlakeIdGenerator;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.io.Serializable;

/**
 * @author yyq
 */
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "sys_menu")
@Table(name = "sys_menu")
public class Menu implements Serializable {

    private static final long serialVersionUID = -5417810396919706043L;

    @Id
    @GeneratedValue(generator = "snowFlake")
    @GenericGenerator(name = "snowFlake", strategy = SnowFlakeIdGenerator.TYPE)
    @Column(nullable = false)
    private Long menuId;

    private String menuCode;

    private String menuName;

    private String menuValue;

    private Long parentId;

    private Integer status;

    private Boolean hidden;

    private Boolean fullScreen;

    private Integer level;

    private Integer sortNum;

    private String icon;

    private String comment;
}
